### PR TITLE
Add tap to view detail annotation on SearchUI map root

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description

Add result detail view for Demo > SearchUI tab to expose additional information from a map annotation

Simulator Screen Recording - iPhone 15 - 2024-07-10 at 12 49 17

### Checklist
- [NA] Update `CHANGELOG`

### CODE REVIEW

File: Sources/Demo/MapRootController.swift [hunk 1]
```diff
@@ -46,10 +46,16 @@ class MapRootController: UIViewController {
...
-        annotationsManager.annotations = results.map {
-            var point = PointAnnotation(coordinate: $0.coordinate)
-            point.textField = $0.name
+        annotationsManager.annotations = results.map { result in
+            var point = PointAnnotation(coordinate: result.coordinate)
+            point.textField = result.name
             UIImage(named: "pin").map { point.image = .init(image: $0, name: "pin") }
+
+            // Present a detail view upon annotation tap
+            point.tapHandler = ((MapContentGestureContext) -> Bool)? { [weak self] _ in
```
Comment (kried) [inline Sources/Demo/MapRootController.swift:55]: Will this work?
```suggestion
            point.tapHandler = { [weak self] _ in
```

File: Sources/Demo/MapRootController.swift [hunk 2]
```diff
@@ -46,10 +46,16 @@ class MapRootController: UIViewController {
...
+                self?.present(result: result)
+                return false
```
Comment (kried) [inline Sources/Demo/MapRootController.swift:57]: Are you sure you need `false` here?
From the docs:
> Should return `true` if the gesture is handled, or `false` to propagate it to the annotations or layers below.

It looks like now you ignore the returned value in present
```suggestion
                return self?.present(result: result) ?? false
```

File: Sources/Demo/ResultDetailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    func numberOfSections(in tableView: UITableView) -> Int {
+        2
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            resultComponents.count
+        default:
+            0
```
Comment (kried) [inline Sources/Demo/ResultDetailViewController.swift:89]: Is a second section then necessary if it shows 0 rows?

Comment (aokj4ck) [reply Sources/Demo/ResultDetailViewController.swift:89]: Ah yes I had an idea to add a second section but never followed-through. Maybe later on!

**Pull request merged**